app/domain/authentication/security/process_request_proxies.rb

```diff
@@ -0,0 +1,58 @@
+require 'command_class'
+
+module Authentication
+  module Security
+    ProcessRequestProxies ||= CommandClass.new(
+      dependencies: {
+        # InputValidation
+      },
+      inputs: [:request]
+    ) do
+
+      def call
+        validate_request
+        process_xff_header
+      end
+
+      private
+
+      Struct.new('RemoteAddress', :client_ip, :proxy_list)
+
+      def validate_request
+        raise ArgumentError, '@request' if @request.nil?
+        raise ArgumentError, '@request.remote_addr' if nil_or_empty @request.remote_addr
+      end
+
+      def process_xff_header
+        remote_addr = @request.remote_addr
+        xff_header = @request.headers['X-Forwarded-For']
```

Comment:
do we want to set `'X-Forwarded-For'` as a class const?

Reply:
Actually it was at the beginning